src/processes/linux_list.rs

```diff
@@ -43,6 +61,80 @@ pub fn active_executables() -> Result<ProcessList> {
     Ok(executables.into_values().collect())
 }
 
+pub fn visible_windows() -> Result<HashSet<PID>> {
+    if env::var("WAYLAND_DISPLAY").is_ok() {
+        wayland_visible_windows()
+    } else if env::var("DISPLAY").is_ok() {
+        x11_visible_windows()
+    } else {
+        Ok(HashSet::new())
+    }
+}
+
+fn wayland_visible_windows() -> Result<HashSet<PID>> {
+    // TODO
+    Ok(HashSet::new())
+}
+
+fn x11_visible_windows() -> Result<HashSet<PID>> {
+    let mut pids: HashSet<PID> = HashSet::new();
+
+    let (conn, screen) = x11rb::connect(None)?;
+    // do we want to handle multiple screens as well?
+    let root = conn.setup().roots[screen].root;
```

Comment:
Our X Server might be handling multiple screens. Do we want to handle the windows on those screens as well?